### DISCUSSION TITLE
Automations Freeze Using Update Row with Looping

### DISCUSSION THREAD

**Discussion by smurraystrata** [Budibase/budibase#8974] [Help] [answered]
I am trying to run an automation that pulls a filtered list of rows (using query rows action with 2 filters) and then updates that list of rows using a looping update row action. I tested the query rows action and it returned the list correctly, but when I add the update row action (using the {{ steps.1.rows.0.id }} binding) the entire page freezes (until I refresh the page, I waited over an hour once) whenever I try to test it and does not show up in the automation logs (I tried setting it with a cron trigger). I have added a limit of 2 on the query rows and the max loop iterations and I have set this automation to work with both the internal db and with an external MySQL db with no change. Any suggestions on what to try further?

Thank you in advance!

Edit: I tried it without any looping and just doing one row and it works without freezing, so it seems to be something with the looping causing the freezing.

**Comment by melohagan** [comment]
Hey @jedman20 

Thanks for raising this. Could you please provide a video recording or step-by-step screenshots and I can try and reproduce.

**Reply by smurraystrata** [reply]
Thank you for looking Melohagan! Here are screenshots of the automation and the columns set up in the table.
![image](https://user-images.githubusercontent.com/120056389/206510290-d2723368-8cc7-4b41-8994-b65d72ca4279.png)
![image](https://user-images.githubusercontent.com/120056389/206510468-2855afe4-b728-4905-9de3-cae1c6f0a08f.png)
![image](https://user-images.githubusercontent.com/120056389/206510614-f5411d77-33da-4f32-98df-c64e3eee3021.png)
![image](https://user-images.githubusercontent.com/120056389/206510662-f05bde5a-3374-48b0-88c5-e672f2d0137b.png)
![image](https://user-images.githubusercontent.com/120056389/206511013-d20cd7bc-451f-4c17-9ac9-7e18e2568e0b.png)

**Comment by melohagan** [comment, marked as answer]
Hey @jedman20 

Thanks for the screenshots! Your looping binding is simply incorrect. It expects an array to loop over, therefore change `{{ steps.1.rows.0.id }}` -> `{{ steps.1.rows }}`

With that said, the app shouldn't crash. I've raised a bug here: https://github.com/Budibase/budibase/issues/8989
